File: 0203_Remove Linked List Elements/RemoveLinkedListElements.java
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null) return head;
        ListNode preHead = new ListNode(-1);
        preHead.next = head;
        ListNode current = preHead;
        
        while(current!=null){
            if(current.next!=null && current.next.val == val){
                current.next = current.next.next;
            }else{
                current = current.next;
            }
        }
        
        return preHead.next;
        
    }
}

File: 0203_Remove Linked List Elements/RemoveLinkedListElements.cs
public class Solution
{
    public ListNode RemoveElements(ListNode head, int val)
    {
        ListNode preHeader = new ListNode(-1);

        ListNode current = preHeader;
        ListNode index = head;

        while (index != null)
        {
            if(index.val == val)
            {
                current.next = index;
                current = current.next;
            }

            index = index.next;
        }

        return preHeader.next;
    }
}
